ScopeView: allow lissajou for more than 2 chs as overlaid pairs

elgiano opened the pull request:
<!-- Please see CONTRIBUTING.md for guidelines. -->

## Purpose and Motivation

Currently ScopeView's lissajou style limits itself to the first two channels being monitored. It is however useful to be able to plot lissajou curves for more than one channel pair, the simplest way being overlaying them. ScopeView is already able to monitor more than two channels, and it has an option to choose different `waveColors`, which make overlaid XY plots discernible, like in the current `overlay` style.

Before making this PR, I've tried composing two stethoscopes in a `StackLayout`, with transparent backgrounds, but it failed at overlaying plots, perhaps due to the QtCollider implementation drawing a bitmap.

This PR doesn't compromise previous functionality: when scoping a 2 channel bus, XY plots are still 2 channels, and default GUIs have options to change number of channels monitored and their offset, so that switching between adjacent channel pairs is still possible the same way as it was before. However, if anyone is relying on ScopeView hiding every other channel than the first two from XY plots, e.g. expecting to see XY plots only for the first 2 channels when monitoring a 6 channels bus, they would now get more confusing plots, especially if they use default colors.

*Possible use cases*:
- inspecting the stereo image of multiple sub-mixes in the same plot
- my real use case: monitoring time-delay phase-space plots of multiple channels in the same plot

*Documentation*:
I've changed the spots where it says that only the first two channels would be plotted

*Tests*:
I couldn't come up with ideas for tests of this new functionality, as we can't easily check how many channels are being plotted in lissajou style.

## Types of changes

- New feature

## To-do list

- [x] Updated documentation
- [x] This PR is ready for review

elgiano commented:
here are some pictures of three overlaid phase plots from my current project. Note that these are three different moments in time, i.e. only two plots are overlaid at a given moment.
![phaseplots](https://github.com/supercollider/supercollider/assets/317018/42a9000d-0722-4fdb-9dc6-245fb9b58413)

JordanHendersonMusic reviewed (commented):
This looks great! Code just extends what is there safety. 

![38538](https://github.com/supercollider/supercollider/assets/54682166/120c59f8-fd87-4f25-bd5b-69584aa97214)

I do think the existing documentation examples are quite verbose making it a little hard to discover this feature. Would you mind adding another really short example demonstrating this? Just something like:
```supercollider
{ 
	var f = { |l,h| SinOsc.kr(0.1, {pi.rand}).range(l, h) } ;
	Pulse.ar({ f.(20, 1000) }!6, { f.(0.0, 1) }!6, { f.(0.0, 1.0) }!6) 
}.play;

c = s.scope;
c.style = 2;
c.numChannels = 6;
c.scopeView.waveColors = [Color.red, Color.red, Color.cyan, Color.cyan, Color.white, Color.white];
```

elgiano commented:
thanks @JordanHendersonMusic, I've added your example to Stethoscope's HelpSource, is it where you envisioned it to be? 
I've just changed the order of operations because I think it results in a clearer progression if they are executed one by one.

JordanHendersonMusic commented:
Did you test that, because I found that setting the colour then changing the style causes the colours to be reset?

Other than that it looks fine, it's just so this feature is a bit more discoverable.

elgiano commented:
Yeah, I had tested it, and on my linux machine colors don't reset. What OS are you on?

JordanHendersonMusic commented:
> Yeah, I had tested it, and on my linux machine colors don't reset. What OS are you on?

Ah my mistakes, its changing the number of channels that causes the colour information to be lost.

elgiano commented:
Oh yeah, that I have noticed as well. Great, we are good to go then! thanks a lot

JordanHendersonMusic reviewed (approved):
Looks great! This PR adds a fully backwards compatible feature by just extending on the c++.
